### PR TITLE
Support result in LoggingInvocationHandler

### PR DESCRIPTION
**Pull request by nevillelyh**
## Description

Support method returned value in `LoggingInvocationHandler`.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/util/LoggingInvocationHandler.java [hunk 1]
```diff
@@ -66,7 +73,12 @@ protected Object handleInvocation(Object proxy, Method method, Object[] args)
             throw t;
         }
         Duration elapsed = Duration.nanosSince(startNanos);
-        logger.accept(format("%s succeeded in %s", invocationDescription(method, args), elapsed));
+        if (includeResult) {
+            logger.accept(format("%s succeeded in %s and returned %s", invocationDescription(method, args), elapsed, result));
```
Comment (kokosing) [inline lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/util/LoggingInvocationHandler.java:77]: use `formatArgument`

**Review by kokosing — APPROVED** [review]
% comment

**Comment by kokosing** [comment]
Is there any places where we leverage this? Maybe there are some interfaces that return simple objects like access controls.

**Comment by kokosing** [comment]
FYI @findepi

**Comment by nevillelyh** [comment]
It's used by 3 classes only.
- `AccessControl` - all methods return `void` and throw exceptions on access denied
- `JdbcClient` - results can be large like list of columns
- `ThriftHiveMetastore.Iface` - large results similar to `JdbcClient`

File: lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/util/LoggingInvocationHandler.java (hunk 1 above)
Comment (nevillelyh) [reply lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/util/LoggingInvocationHandler.java]: Fixed

**Review by kokosing — APPROVED** [review]

**Pull request merged**